core/src/components/item/item.tsx

```diff
@@ -134,8 +136,21 @@ export class Item implements ComponentInterface, AnchorInterface, ButtonInterfac
    */
   @Prop() type: 'submit' | 'reset' | 'button' = 'button';
 
+  /**
+   * A callback used to format the counter text.
+   * By default the counter text is set to "itemLength / maxLength".
+   */
+  @Prop() counterFormatter: CounterFormatter = (itemLength: number, maxLength: number) => `${itemLength} / ${maxLength}`;
+
   @State() counterString: string | null | undefined;
 
+  @Watch('counterFormatter')
+  counterFormatterChanged() {
+    if (this.counter) {
```

Comment:
This is redundant. The `updateCounterOutput` call already have an `this.counter` check